apps/web/src/server/profiles/searchProfiles.ts

```diff
@@ -23,31 +23,53 @@ export const countProfiles = async (
   searchParams: Pick<SearchParams, 'query'>,
   user: Pick<SessionUser, 'id'> | null,
 ) => {
-  const searchTerm = searchToTsQueryInput(searchParams.query)
+  const searchTerm = cleanSearchTerm(searchParams.query) ?? ''
   const userId = user?.id ?? null
 
   const result = await prismaClient.$queryRaw<{ count: number }[]>`
-      SELECT COUNT(DISTINCT users.id)::integer as count
-      FROM users
-      WHERE (
-          ${searchTerm ?? ''} = ''
-              OR to_tsvector('french',
-                             unaccent(coalesce(users.name, '') || ' ' || coalesce(users.location, '') || ' ' ||
-                                      coalesce(users.title, '') || ' ' || coalesce(users.description, ''))) @@
-                 to_tsquery('french', unaccent(${searchTerm}))
-          )
-        AND (
-          /* Authorization*/
-          /* User is public  */
-          users.is_public = true
-              /* User is private and user is self */
-              OR users.id = ${userId}::uuid
-          )
-        AND (
-          users.deleted IS NULL
-          )
+      WITH search AS (SELECT unaccent(${searchTerm}) AS term),
+      profiles AS (SELECT users.id                                                 AS id,
+                                  users.slug                                                AS slug,
+                                  users.created                                             AS created,
+                                  users.last_name                                           AS last_name,
+                                  users.name                                                AS name,
+                                  users.location                                            AS location,
+                                  users.title                                               AS title,
+                                  users.description                                         AS description,
+                                  COUNT(DISTINCT profile_follows.id)                        AS follows_count,
+                                  COUNT(DISTINCT resources.id)                              AS resources_count
+                        FROM users
+                             LEFT JOIN profile_follows ON users.id = profile_follows.profile_id
+                             LEFT JOIN resources ON users.id = resources.created_by_id
```

Comment:
no need for joins and select COUNT(...) for the count query as they are only useful for sorting